### PR TITLE
Upgrade to Wagtail 3.0.3

### PR DESCRIPTION
**Pull request by chosak**
(Note: this PR is going into `upgrade/wagtail`, not `main`.)

This PR updates the version of Wagtail used from 2.16.3 to 3.0.3.

With this PR it's possible to run a local server and load basic pages on the site. Note that some of our Python tests and some of our acceptance tests fail against this PR; see below for details. 

This PR is primarily the code changes necessary to get `tox` to even run completely and report test failures:

- [Updating Wagtail module paths](https://docs.wagtail.org/en/latest/releases/3.0.html#changes-to-module-paths)
- [Removing special-purpose field types](https://docs.wagtail.org/en/latest/releases/3.0.html#id1)

This PR also includes a temporary pin of the wagtail-treemodeladmin package to the branch being worked on in https://github.com/cfpb/wagtail-treemodeladmin/pull/34, as it adds 3.x support.

## Notes and todos

The Python tests that fail here all relate to our custom code for StreamField migrations, which we document [here](https://cfpb.github.io/consumerfinance.gov/migrations/). These fail because of [this documented change](https://docs.wagtail.org/en/latest/releases/3.0.html#replaced-content-json-textfield-with-content-jsonfield-in-pagerevision) to the way PageRevision data is stored. I would prefer to address those failures in another PR; see internal Design-and-Content-Team#117 for some context.

Besides the additional few functional test failures, there are definitely additional minor changes we'll want to make related to the Wagtail 3.0 upgrade; for example updating [documentation](https://cfpb.github.io/consumerfinance.gov/wagtail-pages/) that refers to the old Wagtail module paths, or deprecating our custom page description functionality, which we backported from 3.0 in #7134. I'd like to get this PR merged into `upgrade/wagtail` to do the heavy lifting and then address those things in subsequent smaller PRs.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets

### CODE REVIEW

File: requirements/libraries.txt [hunk 1]
```diff
@@ -33,7 +33,8 @@ wagtail-flags==5.2.0
 wagtail-inventory==1.6
 wagtail-placeholder-images==0.1.1
 wagtail-sharing==2.7
-wagtail-treemodeladmin==1.5.0
+git+https://github.com/cfpb/wagtail-treemodeladmin@wag3-support#egg=wagtail-treemodeladmin
+# wagtail-treemodeladmin==1.5.0
```
Comment (willbarton) [inline requirements/libraries.txt:37]:
```suggestion
wagtail-treemodeladmin==1.6.0
```

Comment (chosak) [reply requirements/libraries.txt:37]: How about doing this on `main` instead to reduce the size of the Wagtail upgrade effort? Then we can rebase `upgrade/wagtail`.

Comment (chosak) [reply requirements/libraries.txt:37]: Thanks @willbarton - now that this upgrade is merged into main in https://github.com/cfpb/consumerfinance.gov/pull/7445, I'm going to rebase `upgrade/wagtail`.

Comment (chosak) [reply requirements/libraries.txt]: Done, and rebased this PR branch against that.

File: test/cypress/integration/admin/admin-helpers.cy.js [hunk 2]
```diff
@@ -142,6 +142,7 @@ export class AdminPage {
   clickBlock(name) {
     return cy
       .get(`.action-add-block-${name}`, { timeout: 60000 })
+      .scrollIntoView()
```
Comment (anselmbradford) [inline test/cypress/integration/admin/admin-helpers.cy.js:145]: QQ here. I noticed in [the docs](https://docs.cypress.io/api/commands/scrollintoview) it says:

> It is [unsafe](https://docs.cypress.io/guides/core-concepts/retry-ability#Only-queries-are-retried) to chain further commands that rely on the subject after .scrollIntoView().

Should this perhaps instead be something like?

```js
cy.get(`.action-add-block-${name}`, { timeout: 60000 })
    .scrollIntoView()
    .should('be.visible');
return cy.get(`.action-add-block-${name}`, { timeout: 60000 }).click();
```

Comment (chosak) [reply test/cypress/integration/admin/admin-helpers.cy.js]: Thanks @anselmbradford, nice find. Hopefully addressed satisfactorily in 7a140d0.

**Review by willbarton — APPROVED** [review]

**Pull request merged**